src/track/heaptrack.sh.cmake

```diff
@@ -250,17 +258,27 @@ else
         --eval-command="run" --args "$client" "$@"
   else
     echo "injecting heaptrack into application via GDB, this might take some time..."
+    case $(uname) in
+        Linux*)
+            dlopen=__libc_dlopen_mode
+        ;;
+        FreeBSD*)
+            dlopen=dlopen@plt
```

Comment:
just a question: will this work with executables that don't link against `-ldl` directly? on linux at least, the `dlopen` symbol resides in that library so if the executable isn't linked against that, heaptrack would fail. but the mysterious libc symbol is always available, which is why I do this here.

Reply:
On FreeBSD, `dlopen` resides in the `rtld`.

`libc` and `libdl` (the latter being a compatibility dummy pretty much) just weakly reference it.

```
/usr/src % readelf -a /usr/lib/libdl.so.1 | rg dlopen
    11: 0000000000001b70    22 FUNC    WEAK   DEFAULT   12 dlopen@@FBSD_1.0
    12: 0000000000001c40    22 FUNC    WEAK   DEFAULT   12 fdlopen@@FBSD_1.3
/usr/src % readelf -a /lib/libc.so.7 | rg dlopen
0000001cb500  053000000007 R_X86_64_JUMP_SLO 000000000019dd90 dlopen@@FBSD_1.0 + 0
   373: 000000000019dff0    22 FUNC    WEAK   DEFAULT   12 fdlopen@@FBSD_1.3
  1328: 000000000019dd90    22 FUNC    WEAK   DEFAULT   12 dlopen@@FBSD_1.0
  2117: 000000000019e050     8 FUNC    WEAK   DEFAULT   12 _rtld_is_dlopened@@FBSDprivate_1.0
/usr/src % readelf -a /libexec/ld-elf.so.1 | rg dlopen
     4: 000000000000ab30    16 FUNC    GLOBAL DEFAULT   11 fdlopen@@FBSD_1.3
    14: 000000000000a990    14 FUNC    GLOBAL DEFAULT   11 dlopen@@FBSD_1.0
    18: 000000000000ca20   234 FUNC    GLOBAL DEFAULT   11 _rtld_is_dlopened@@FBSDprivate_1.0
```

Reply:
cool thanks for the info